docs/usage/operations.rst

```diff
@@ -621,3 +621,100 @@ Class Definitions
                   ),
               ),
           ]
+
+
+.. py:class:: SaferAddCheckConstraint(model_name: str, constraint: models.CheckConstraint)
+
+    Provides a safer way to add a check constraint to an existing model.
+
+    :param model_name: Model name in lowercase without underscores.
+    :type model_name: str
+    :param constraint: The object representing the constraint to add.
+    :type constraint: models.CheckConstraint
+
+    **Why use this SaferAddCheckConstraint operation?**
+    ---------------------------------------------------
+
+    When using Django's default ``AddConstraint`` operation, the SQL created
+    has the following form:
+
+    .. code-block:: sql
+
+      ALTER TABLE foo
+      ADD CONSTRAINT bar_not_negative
+      CHECK (bar >= 0);
+
+    This operation acquires an AccessExclusive lock, which is the most
+    constricted lock in Postgres, blocking any reads, writes, maintanence
+    activities, and other schema changes on the table.
+
+    It will also scan the whole table to make sure there are no violations of
+    the new constraint. All that while holding onto that lock.
+
+    This safer operation, will insted perform the following:
+
+    .. code-block:: sql
+
+      -- Add a NOT VALID constraint.
+      -- This type of constraint still works, but only for new writes.
+      -- It still requires the AccessExclusive lock, but as it doesn't need to
+      -- scan the table, it runs very fast.
+      ALTER TABLE foo
+      ADD CONSTRAINT bar_not_negative
+      CHECK (bar >= 0)
+      NOT VALID;
+
+      -- Validate the constraint.
+      -- This operation needs to scan the table, but it only holds a
+      -- ShareUpdateExclusive lock, which won't block reads or writes.
+      ALTER TABLE foo VALIDATE CONSTRAINT bar_not_negative;
+
```

Comment:
Is it worth adding something about why `atomic = False` (mentioned below) is necessary? That is, that to avoid holding the exclusive lock while the subsequent validation happens, they need to not be in a transaction together?

(My initial thought was that requiring `atomic = False` was unnecessary until I thought about this.)

Reply:
Fixup commit: e7793abde9b20ae69c9cca64754af812cef2e94f